### PR TITLE
3830 Fix margin size in CropForeground transform

### LINKED ISSUE #3830: CropForeground pads asymmetric
**Describe the bug**
In `CropForeground`/`CropForegroundd` the user can request a margin around the foreground voxels. However, currently the margin is only correctly padded on the upper side of a dimension.

See:
https://github.com/Project-MONAI/MONAI/blob/c3db337644d9525610d9fab3cfd30f963438dc6a/monai/transforms/utils.py#L924-L926

**To Reproduce**
```python
from monai.transforms import CropForeground
import numpy as np

transform = CropForeground(margin=2)
arr = np.eye(3)[np.newaxis, ...]
for i in range(3):
    inp = arr[:, i]
    res = transform(inp)
    print(inp, res.shape)
# [[1. 0. 0.]] (1, 3)
# [[0. 1. 0.]] (1, 4)
# [[0. 0. 1.]] (1, 5)
```

**Expected behavior**
The specified margin should be enforced by cropping or padding on both ends of a dimension. 

**Environment**
Monai 0.8.0, but see Permalink above

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3830 .

### Description
This PR fixed the margin size issue in `CropForeground` transform and added flag and unit tests.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utils.py [hunk 1]
```diff
@@ -891,16 +892,20 @@ def generate_spatial_bounding_box(
         [1st_spatial_dim_start, 2nd_spatial_dim_start, ..., Nth_spatial_dim_start],
         [1st_spatial_dim_end, 2nd_spatial_dim_end, ..., Nth_spatial_dim_end]
 
-    The bounding boxes edges are aligned with the input image edges.
-    This function returns [-1, -1, ...], [-1, -1, ...] if there's no positive intensity.
+    If `align_image_edge`, the bounding boxes edges are aligned with the input image edges.
+    This function returns [0, 0, ...], [0, 0, ...] if there's no positive intensity.
```
Comment (wyli) [inline monai/transforms/utils.py:896]: https://github.com/Project-MONAI/MONAI/blob/fd63af4c9082854bc4ad0b01990b3747aa77970e/monai/transforms/croppad/array.py#L1186 this also use the util

Comment (Nic-Ma) [reply monai/transforms/utils.py:896]: Hi @wyli ,

I added it to this transform and the DeepGrow transform in the latest commit.

Thanks.

**Review by wyli — APPROVED** [review]

**Comment by Nic-Ma** [comment]
Updated arg name to `allow_smaller` to be consistent with:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L874

Thanks.

**Pull request merged**